ai-quick-actions/deploy-with-smc.md

```diff
@@ -0,0 +1,258 @@
+HUGGINGFACE_TOKEN =  "<HUGGINGFACE_TOKEN>" # Your huggingface token
+!huggingface-cli login --token $HUGGINGFACE_TOKEN 
+```
+
+[This](https://huggingface.co/docs/huggingface_hub/guides/download#download-an-entire-repository) provides more information on using `snapshot_download()` to download an entire repository at a given revision. Models in the HuggingFace hub are stored in their own repository.
+
+```python
+# Download the ELYZA model from Hugging Face to a local folder.
+#
+
+from huggingface_hub import snapshot_download
+from tqdm.auto import tqdm
+
+model_name = "elyza/ELYZA-japanese-Llama-2-13b-instruct"
+local_dir = "models/ELYZA-japanese-Llama-2-13b-instruct"
+
+snapshot_download(repo_id=model_name, local_dir=local_dir, force_download=True, tqdm_class=tqdm)  
+
+print(f"Downloaded model {model_name} to {local_dir}")
+```
+
+## Upload Model to OCI Object Storage
+
+```python
+model_prefix = "ELYZA-japanese-Llama-2-13b-instruct/" #"<bucket_prefix>" 
+bucket= "<bucket_name>" # this should be a versioned bucket
+namespace = "<bucket_namespace>"
+
+!oci os object bulk-upload --src-dir $local_dir --prefix $model_prefix -bn $bucket -ns $namespace --auth "resource_principal" 
+```
+
+## Create Model by Reference using ADS
+
+```python
+from ads.model.datascience_model import DataScienceModel
+ 
+artifact_path = f"oci://{bucket}@{namespace}/{model_prefix}"
+ 
+model = (DataScienceModel()
+  .with_compartment_id(compartment_id)
+  .with_project_id(project_id)
+  .with_display_name("ELYZA-japanese-Llama-2-13b-instruct")
+  .with_artifact(artifact_path)
+)
+ 
+model.create(model_by_reference=True)
+```
+
+### Import Model Deployment Modules
+
+```python
+from ads.model.deployment import (
+    ModelDeployment,
+    ModelDeploymentContainerRuntime,
+    ModelDeploymentInfrastructure,
+    ModelDeploymentMode,
+)
+```
+
+### Setup Model Deployment Infrastructure
+
+```python
+infrastructure = (
+    ModelDeploymentInfrastructure()
+    .with_project_id(project_id)
+    .with_compartment_id(compartment_id)
+    .with_shape_name(instance_shape)
+    .with_bandwidth_mbps(10)
+    .with_replica(1)
+    .with_web_concurrency(10)
+    .with_access_log(
+        log_group_id=log_group_id,
+        log_id=log_id,
+    )
+    .with_predict_log(
+        log_group_id=log_group_id,
+        log_id=log_id,
+    )
+)
+```
+
+### Configure Model Deployment Runtime
+
+```python
+env_var = {
+    'BASE_MODEL': model_prefix, 
+    'PARAMS': '--served-model-name odsc-llm --seed 42', 
+    'MODEL_DEPLOY_PREDICT_ENDPOINT': '/v1/completions', 
+    'MODEL_DEPLOY_ENABLE_STREAMING': 'true'
+}
+
+container_runtime = (
+    ModelDeploymentContainerRuntime()
+    .with_image(container_image)
+    .with_server_port(8080)
+    .with_health_check_port(8080)
+    .with_env(env_var)
+    .with_deployment_mode(ModelDeploymentMode.HTTPS)
+    .with_model_uri(model.id)
+    .with_region(region)
+    .with_overwrite_existing_artifact(True)
+    .with_remove_existing_artifact(True)
```

Comment:
Probably we don't need this. It will remove the original artifacts form OS bucket. Or maybe we need explicitly mention this?